### PR TITLE
[16.0][ADD] sale_tier_validation_block_print

### PR DESCRIPTION
**Pull request by AungKoKoLin1997**
This module blocks the printing of the sale order report if the order is not yet approved. It prevents both manual printing from the print action and printing when clicking `SEND BY EMAIL`. 

@qrtl QT3715

### CODE REVIEW

File: sale_tier_validation_block_print/reports/sale_report.py [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+    _name = "report.sale.report_saleorder"
+    _description = "Sale Order Report"
+
+    @api.model
+    def _get_report_values(self, docids, data=None):
+        docs = self.env["sale.order"].browse(docids)
+        for order in docs:
+            if order.need_validation or (order.review_ids and not order.validated):
+                raise ValidationError(
+                    _("This operation is blocked before the order is approved.")
```
Comment (yostashiro) [inline sale_tier_validation_block_print/reports/sale_report.py]: We can make the message more explicit here.
```suggestion
                    _("Quotation printing is blocked until the order is approved.")
```

File: sale_tier_validation_block_print/tests/test_sale_tier_validation_block_print.py [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+            }
+        )
+
+        report = self.env["report.sale.report_saleorder"]
+        # Attempt to render the report before validation
+        with self.assertRaises(ValidationError):
+            report._get_report_values(docids=[so.id])
+        # Perform validation process
+        so.order_line.price_unit = 45
+        so.request_validation()
```
Comment (yostashiro) [inline sale_tier_validation_block_print/tests/test_sale_tier_validation_block_print.py]: No need to have a step to change the price, but we want to make sure that printing is still blocked after a validation request.
```suggestion
        so.request_validation()
        with self.assertRaises(ValidationError):
            report._get_report_values(docids=[so.id])
```

**Review by yostashiro — CHANGES_REQUESTED** [review]

**Review by yostashiro — APPROVED** [review]
Code review.

**Review by kanda999 — APPROVED** [review]
Functional review

**Comment by yostashiro** [comment]
Got this comment https://x.com/360ERP_Cas/status/1820921116279390285. Should we try to enhance sale_tier_validation instead?

CC: @BT-cserra @bizzappdev @StefanRijnhart @rousseldenis @pedrobaeza

**Comment by bosd** [comment]
👍 To enhance sale_tier_validation

**Comment by bizzappdev** [comment]
IMO, it will be best to integrate into an original module with a setting in config/system parameters for blocking reports.

**Comment by AungKoKoLin1997** [comment]
This PR is superseded by https://github.com/OCA/sale-workflow/pull/3269.

**Pull request closed without merging**